src/parser.js

```diff
@@ -994,18 +994,22 @@ export default class Parser {
           this.withPath(contentType, () => {
             let formattedResponseBody = responseBody;
 
-            if (typeof(responseBody) !== 'string') {
-              formattedResponseBody = JSON.stringify(responseBody, null, 2);
-            }
+            try {
+              if (typeof(responseBody) !== 'string') {
+                formattedResponseBody = JSON.stringify(responseBody, null, 2);
+              }
 
-            const bodyAsset = new Asset(formattedResponseBody);
-            bodyAsset.classes.push('messageBody');
+              const bodyAsset = new Asset(formattedResponseBody);
+              bodyAsset.classes.push('messageBody');
 
-            if (this.generateSourceMap) {
-              this.createSourceMap(bodyAsset, this.path);
-            }
+              if (this.generateSourceMap) {
+                this.createSourceMap(bodyAsset, this.path);
+              }
 
-            response.content.push(bodyAsset);
+              response.content.push(bodyAsset);
+            } catch (exception) {
+              this.createAnnotation(annotations.DATA_LOST,  this.path, 'Circular references in examples are not yet supported.');
```

Comment:
Yeah, the `createSourceMap` might become an issue. Can you try catch only the JSON stringily please?